### ISSUE TITLE
Handle Scopes appropriately

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/ddf-core-poc#34]
The Engine should be able to handle DDF-specific scopes. When a class is specified in the Scope, the engine should apply the rules to the items within that class.

**Comment by rbakercdisc1** [comment]
There is a red line when you add Entities and Attributes to the Rule Editor. I will delete the other issue.

**Comment by ASL-rmarshall** [comment]
I agree that `Include` and `Exclude` in `Scope` need to be applied for `Entities`. At the moment, the rule is applied to all "datasets" (i.e., sheets) included in the test data Excel file.  For rules that cross-check between entities, it's likely that information from one entity will be processed using `Operations` to be made available via an operation variable for checking of the primary entity dataset.  In this situation, correct handling of `Scope` is necessary to prevent the rule being executed for the entity that is intended to be referenced only via `Operations`.

**Comment by ASL-rmarshall** [comment]
It should also be possible to define "All" for `Scope` / `Entities` / `Include` for generic rules that are based on variable existence like DDF00010 (#94) and DDF00014 (#93).
